### PR TITLE
Increase user login field to 255 characters

### PR DESCRIPTION
**Pull request by ghost**
https://mantis.ilias.de/view.php?id=29502

### CODE REVIEW

File: setup/sql/dbupdate_04.php [hunk 1]
```diff
@@ -24790,4 +24790,14 @@ function writeCtrlClassEntry(ilPluginSlot $slot, array $plugin_data)
     }
     $setting->set('sty_media_cont_mig', 2);
 }
+?>
+<#5428>
+<?php
+global $DIC;
+$DIC->database()->modifyTableColumn("usr_data", "login", [
```
Comment (mjansenDatabay) [inline setup/sql/dbupdate_04.php]: @sragfw We MUST NOT add additional database update steps to `dbupdate_04.php`. Please add this schema change to the corresponding hotfix file for ILIA 5.4.x

Comment (ghost) [reply setup/sql/dbupdate_04.php]: Right, thanks for your hint

**Comment by smeyer-ilias** [comment]
Thanks, will merged to other hotfix files.

File: Services/User/classes/class.ilUserProfile.php [hunk 2]
```diff
@@ -47,7 +47,7 @@ class ilUserProfile
         "username" => array(
                         "input" => "login",
                         "maxlength" => 64,
```
Comment (ghost) [inline Services/User/classes/class.ilUserProfile.php:49]: @smeyer-ilias
Sorry, I found a missing needed change, the max length need also be increased
Could you add this too?

**Comment by smeyer-ilias** [comment]
Yep, done

**Comment by smeyer-ilias** [comment]
We have to reduce the column length due to https://mantis.ilias.de/view.php?id=29625
I 've added this report to the next jour fix agenda since the maximum column size for indexed table columns is currently not defined.

**Comment by smeyer-ilias** [comment]
@sragfw : I reduced the login column to 190 chars, which should support our minimum index length (innodb engine with utf8mb4 and innodb_large_index = 0).

**Comment by ghost** [comment]
190 chars are ok for the client

**Pull request merged**